GVRf/Framework/framework/src/main/jni/engine/renderer/render_state.h

```diff
@@ -0,0 +1,712 @@
+                dest_blend = OneMinusSourceAlpha;
+                stencil_fail = Keep;
+                depth_fail = Keep;
+                stencil_pass = Keep;
+                stencil_func = Always;
+                stencil_func_mask = ~0;
+                stencil_mask = ~0;
+                stencil_ref = 0;
+                pad = 0;
+            }
+
+            int GLBlendFunc(int GVRblendFunc) const
+            {
+                if (GVRblendFunc >= SourceColor)
+                {
+                    return GVRblendFunc + GL_SRC_COLOR - SourceColor;
+                }
+                else
+                {
+                    return GVRblendFunc;
+                }
+            }
+
+            int GVRBlendFunc(int glBlendFunc)
+            {
+                if (glBlendFunc >= GL_SRC_COLOR)
+                {
+                    return glBlendFunc - GL_SRC_COLOR + SourceColor;
+                }
+                else
+                {
+                    return glBlendFunc;
+                }
+            }
+            int GLStencilOp(int GVRstencilOp) const
+            {
+                static int gl_values[8] =
+                {
+                        GL_ZERO, GL_KEEP, GL_REPLACE, GL_INCR,
+                        GL_DECR, GL_INVERT, GL_INCR_WRAP, GL_DECR_WRAP
+                };
+                return gl_values[GVRstencilOp];
+            }
+
+            int GLStencilFunc() const
+            {
+                return stencil_func + GL_NEVER;
+            }
+
+            int GVRStencilFunc(int GLstencilFunc) const
+            {
+                return GLstencilFunc - GL_NEVER;
+            }
+
+            int GVRStencilOp(int GLstencilOp) const
+            {
+                switch (GLstencilOp)
+                {
+                    case GL_ZERO: return Zero;
+                    case GL_REPLACE: return Replace;
+                    case GL_INCR: return Increment;
+                    case GL_DECR: return Decrement;
+                    case GL_INVERT: return Invert;
+                    case GL_INCR_WRAP: return IncrWrap;
+                    case GL_DECR_WRAP: return DecrWrap;
+                    case GL_KEEP: return Keep;
+                    default: return Keep;
+                };
+            }
+        };
+
+        RenderFlags render_flags;
+        int render_order;
+        float offset_factor;
+        float offset_units;
+        float sample_coverage;
+
+    public:
+
+        void init()
+        {
+            render_flags.init();
+            render_order = Geometry;
+            sample_coverage = 1.0f;
+            offset_factor = 0;
+            offset_units = 0;
+        }
+
+        RenderModes& operator=(const RenderModes& src)
+        {
+            render_flags = src.render_flags;
+            offset_factor = src.offset_factor;
+            offset_units = src.offset_units;
+            render_order = src.render_order;
+            sample_coverage = src.sample_coverage;
```

Comment:
Missing return; please compile with GVRF_USE_CLANG=true and address the warnings in the new and changed code.